packages/hint-valid-doctype/src/hint.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * @fileoverview Hint to validate if the doctype is correct
+ */
+import { Category } from 'hint/dist/src/lib/enums/category';
+import { HintScope } from 'hint/dist/src/lib/enums/hintscope';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+import { IHint, HintMetadata, FetchEnd, IAsyncHTMLElement, ResponseBody } from 'hint/dist/src/lib/types';
+import { debug as d } from 'hint/dist/src/lib/utils/debug';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class implements IHint {
+
+    public static readonly meta: HintMetadata = {
+        docs: {
+            category: Category.interoperability,
+            description: `Hint to validate if the doctype is correct`
+        },
+        id: 'valid-doctype',
+        schema: [],
+        scope: HintScope.any
+    }
+
+    public constructor(context: HintContext) {
+        const validDoctype = '<!doctype html>';
+        const doctypeRegex = /(<!doctype ).+(>)(.+)?/;
+        const doctypeRegexFactory = (flags?: string) => new RegExp('(<!doctype ).+(>)(.+)?', flags ? flags : 'g');
+
+        const checkDoctypeIsValid = async (resource: string, element: IAsyncHTMLElement, content: string) => {
+            debug(`Checking if the doctype is valid.`);
+
+            const matched = content.match(doctypeRegexFactory('gi'));
+
+            if (!matched || matched.length < 1) {
+                await context.report(resource, element, `The file does not contain a doctype tag`);
+                return;
+            }
+
+            const [contentDoctype] = matched;
+
+            if (contentDoctype.toLowerCase() !== validDoctype) {
```

Comment:
`<!doctype   html   >` is also valid and should be allowed.

(see: https://html.spec.whatwg.org/multipage/syntax.html#the-doctype).

Reply:
We don't do that for other hints. If the value is valid and the intended one, we don't complain (e.g.: [1](https://github.com/webhintio/hint/blob/e7887d4068f20caf8a55b534a3d1e0e72e2f5c1f/packages/hint-meta-charset-utf-8/src/hint.ts#L108)).